packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -1049,8 +1055,8 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
     }
   }
 
-  _dateAllowed(date, min = this.minDate, max = this.maxDate) {
-    return (!min || date >= min) && (!max || date <= max);
+  _dateAllowed(date, min = this.minDate, max = this.maxDate, isDateAvailable = this.isDateAvailable) {
+    return (!min || date >= min) && (!max || date <= max) && (!isDateAvailable || isDateAvailable(date))
```

Comment:
I would suggest to reuse the `typeof` check here, as it's more precise than checking for a falsy value.